### PR TITLE
[prometheus-node-exporter] add automountServiceAccountToken for serviceaccount resource

### LINKED ISSUE #4628: [prometheus-node-exporter] Allow option to set automountServiceAccountToken on serviceAccount resource
### Is your feature request related to a problem ?

Its not possible to set automountServiceAccountToken on serviceaccount resource.

### Describe the solution you'd like.

Add option for `automountServiceAccountToken` on serviceaccount.

### Describe alternatives you've considered.

NA

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by rgaduput**
#### What this PR does / why we need it
Option to set automountServiceAccountToken on service account resource.

#### Which issue this PR fixes

- fixes #4628 

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-node-exporter/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.36.0
+version: 4.36.1
```
Comment (zeritti) [inline charts/prometheus-node-exporter/Chart.yaml:9]: Please, raise the minor version instead.
```suggestion
version: 4.37.0
```

File: charts/prometheus-node-exporter/templates/serviceaccount.yaml [hunk 2]
```diff
@@ -10,6 +10,7 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
+automountServiceAccountToken: {{ .Values.serviceAccount.automountServiceAccountToken }}
```
Comment (zeritti) [inline charts/prometheus-node-exporter/templates/serviceaccount.yaml:13]: Please, enter the new field in values.yaml assigning it a default value.

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @rgaduput, for your PR. Please, see my comments below.

File: charts/prometheus-node-exporter/templates/serviceaccount.yaml (hunk 2 above)
Comment (rgaduput) [reply charts/prometheus-node-exporter/templates/serviceaccount.yaml:13]: @zeritti its already exists in values.yaml file. So I am just using it in the serviceaccount.yaml
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-node-exporter/values.yaml#L292

Comment (zeritti) [reply charts/prometheus-node-exporter/templates/serviceaccount.yaml:13]: Indeed, apologies, I missed it.

**Review by zeritti — APPROVED** [review]
Thank you, @rgaduput, LGTM.

**Pull request merged**